### DISCUSSION TITLE
Local runtime hosting details.

### DISCUSSION THREAD

**Discussion by flibbertigibbet-uk** [microsoft/aspire#978] [Q&A] [answered]
My background: I did a few months of C# Dapr development two years ago and now I am currently deploying a **non** Dapr  .Net Core distributed system to Azure Container Apps. Aspire seems like a natural fit for my current development activities.

To help me understand how Aspire relates to the above:

1. Could Microsoft clarify if Aspire weaves something like Dapr side-cars into the runtime architecture? I am hoping that Aspire can operate without side-cars though I appreciated that some teams will elect to target Dapr on the cloud from their Aspire VS solution.
2. When an Aspire solution is run locally as part of the code/debug cycle is the VS2022 Aspire tooling building Linux containers under the covers and then launching these containers within the Windows Subsystem for Linux (WSL2)?

**Comment by davidfowl** [comment, marked as answer]
> Could Microsoft clarify if Aspire weaves something like Dapr side-cars into the runtime architecture? I am hoping that Aspire can operate without side-cars though I appreciated that some teams will elect to target Dapr on the cloud from their Aspire VS solution.

By default, not it does not weave anything like dapr side cars in the default experience. Dapr is an API extension that can be enabled. See the example here https://github.com/dotnet/aspire-samples/blob/main/samples/AspireWithDapr/AspireWithDapr.AppHost/Program.cs. Currently there's no deployment support for dapr to ACA but that's being [worked on](https://github.com/dotnet/aspire/pull/862).

> When an Aspire solution is run locally as part of the code/debug cycle is the VS2022 Aspire tooling building Linux containers under the covers and then launching these containers within the Windows Subsystem for Linux (WSL2)?

No it runs your projects as executables, not containers. If you run a container, it'll use docker to run those individual containers.
